plugins/cryptolib/cryptolib.cpp

```diff
@@ -411,6 +411,52 @@ void verifyPKIAlgorithm(const char * pkAlgorithm)
 }
 
 
+//---------------------------------------------------------
+// Helper that reads the given logical file into a string buffer
+// Throws if user does not have read permission or if FNF
+//---------------------------------------------------------
+void loadLFS(const char * lfs, IUserDescriptor * user, StringBuffer &sb)
+{
+    CDfsLogicalFileName lfn;
+    lfn.set(lfs);
+    try
+    {
+        Owned<IDistributedFile> df = queryDistributedFileDirectory().lookup(lfn, user, false, false, false, nullptr, defaultPrivilegedUser);//scope checks
+        if (!df)
+        {
+            throw MakeStringException(-1, "File Not Found");
+        }
+
+        unsigned numParts = df->numParts();
+        if (numParts != 1)
+        {
+            throw MakeStringException(-1, "Part count (%d) is incorrect", numParts);
+        }
+
+        IDistributedFilePart *part = df->getPart(0);
+
+        RemoteFilename rfn;
+        Owned<IFile> file = createIFile(part->getFilename(rfn));
+        if (!file->exists())
+        {
+            throw MakeStringException(-1, "File Not Found");
+        }
+
+        Owned<IFileIO> io = file->open(IFOread);
+        offset_t len = file->size();
+        char * buff = sb.reserve(len);
+        size32_t read = io->read(0, len, buff);
+        assertex(read == len);
+        io->close();
+    }
+    catch(IException * e)
+    {
+        StringBuffer s;
+        VStringBuffer sb("Error accessing Key file '%s' : %s", lfs, e->errorMessage(s).str());
+        throw MakeStringException(-1, "%s", sb.str());
```

Comment:
leaks 'e'. Should Release() before throwing new exception

Reply:
Good catch (haha, see what I did there)

Reply:
trivial: makeStringExceptionV preferred

Reply:
or makeStringException(-1, sb.str())